plugins/modules/zos_apf.py

```diff
@@ -520,29 +520,27 @@ def main():
     operRc = ret.rc
     result['stderr'] = operErr
     result['rc'] = operRc
+    result['stdout'] = operOut
     if operation == 'list':
-        try:
-            dsRx = ""
-            volRx = ""
-            if library:
-                dsRx = re.compile(library)
-            if volume:
-                volRx = re.compile(volume)
-            if sms:
-                sms = "*SMS*"
-            if dsRx or volRx or sms:
+        if not library:
+            library = ""
+        if not volume:
+            volume = ""
+        if sms:
+            sms = "*SMS*"
+        if library or volume or sms:
+            try:
                 data = json.loads(operOut)
-                operOut = ""
-                for d in data[2:]:
-                    ds = d.get('ds')
-                    vol = d.get('vol')
-                    if (dsRx and dsRx.match(ds)) or (volRx and volRx.match(vol)) or (sms and sms == vol):
-                        operOut = operOut + "{0} {1}\n".format(vol, ds)
-        except Exception:
-            pass
-
-    result['stdout'] = operOut
-
+            except json.JSONDecodeError:
+                module.exit_json(**result)
+            for d in data[2:]:
+                ds = d.get('ds')
+                vol = d.get('vol')
+                try:
+                    if (re.match(library, ds)) or (re.match(volume, vol)) or (sms and sms == vol):
```

Comment:
i'm questioning if the effect of the logic is the same as from the other (maybe it's handled elsewhere and i just missed it), 
the old:
```if (dsRx and dsRx.match(ds)) or (volRx and volRx.match(vol)) or (sms and sms == vol):```
vs the new:
```if (re.match(library, ds)) or (re.match(volume, vol)) or (sms and sms == vol):```

just a few snippets of python:
```python
>>> import re
>>> bool(re.match("", "SOME.DATA.SET"))
True
```
```python
>>> bool("")
False
```
```python3
>>> bool("" and re.match("", "SOME.DATA.SET"))
False 
```

My concern is that there may be some cases where a regex match is reported but an empty string was matched rather than a real value. To me, the old ensures that there's a match with _real_ data by having the two part check per conditional clause `dsRX and dsRX.match(ds)`. I think there could be something similar added to the new, maybe something like `(if library and re.match(library, ds)) or ...`.

Reply:
Nice finding, this is an issue bc if `library == ""`, then resulting match will be 
```
>>> bool(re.match("", "SOME.DATA.SET"))
True
```

Will modify this to honor the current behavior.